src/PlottingInterface_TEST.cc

```diff
@@ -0,0 +1,168 @@
+    EXPECT_EQ(fields["pose-position-y"]->ChartCount(), 1);
+
+    // charts test
+    EXPECT_TRUE(fields["pose-position-x"]->Charts().find(1) !=
+            fields["pose-position-x"]->Charts().end());
+    EXPECT_TRUE(fields["pose-position-x"]->Charts().find(2) !=
+            fields["pose-position-x"]->Charts().end());
+    EXPECT_TRUE(fields["pose-position-y"]->Charts().find(1) !=
+            fields["pose-position-x"]->Charts().end());
+
+    // test the removing of the field if it has not attatched charts
+    topic.UnRegister("pose-position-y", 1);
+    EXPECT_EQ(topic.FieldCount(), 1);
+
+
+    // =========== Callback Test ============
+    topic.Register("pose-position-z", 1);
+
+    // update the fields
+    topic.Callback(msg);
+
+    fields = topic.Fields();
+
+    EXPECT_EQ(static_cast<int>(fields["pose-position-x"]->Value()), 10);
+    EXPECT_EQ(static_cast<int>(fields["pose-position-z"]->Value()), 15);
+}
+
+TEST(PlottingInterfaceTest, Transport)
+{
+    // =========== Publish Test =================
+    transport::Node node;
+
+    auto pub = node.Advertise<msgs::Collision> ("/collision_topic");
+    std::this_thread::sleep_for(std::chrono::milliseconds(200));
+
+    auto transport = Transport();
+    transport.Subscribe("/collision_topic", "pose-position-x", 1);
+    transport.Subscribe("/collision_topic", "pose-position-z", 1);
+
+    // prepare the msg
+    msgs::Collision msg;
+    auto pose = new ignition::msgs::Pose();
+    auto vector3d = new ignition::msgs::Vector3d();
+    vector3d->set_x(10);
+    vector3d->set_z(15);
+    pose->set_allocated_position(vector3d);
+    msg.set_allocated_pose(pose);
+
+    // publish to call the topic::Callback
+    pub.Publish(msg);
+
+    // wait for callback
+    std::this_thread::sleep_for(std::chrono::milliseconds(50));
+
+    auto topics = transport.Topics();
+
+    EXPECT_EQ(topics["/collision_topic"]->FieldCount(), 2);
+
+    auto fields = topics["/collision_topic"]->Fields();
+
+    EXPECT_EQ(static_cast<int>(fields["pose-position-x"]->Value()), 10);
+    EXPECT_EQ(static_cast<int>(fields["pose-position-z"]->Value()), 15);
+
+
+    // =========== Many Topics Test =================
+    // add another topic to the transport and subscribe to it
+    node.Advertise<msgs::Int32> ("/test_topic");
+    transport.Subscribe("/test_topic", "data", 2);
+
+    topics = transport.Topics();
+
+    ASSERT_EQ(static_cast<int>(topics.size()), 2);
+    EXPECT_EQ(topics["/test_topic"]->FieldCount(), 1);
+
+
+    // =========== UnSubscribe Test =================
+
+    // test the deletion of the topic if it has no fields
+    transport.Unsubscribe("/collision_topic", "pose-position-z", 1);
+    transport.Unsubscribe("/collision_topic", "pose-position-x", 1);
+
+    topics = transport.Topics();
+    EXPECT_EQ(static_cast<int>(topics.size()), 1);
+}
+
+TEST(PlottingInterfaceTest, PlottingIface)
+{
+//    auto PlotIface = new ignition::gui::PlottingInterface();
```

Comment:
What are you trying to test here? Anything we can help with?

Reply:
that was an old problem, but it tests small functionality, and i think it can be removed